pkg/codegen/go/gen_program.go

```diff
@@ -584,7 +587,64 @@ func (g *generator) genLocalVariable(w io.Writer, v *hcl2.LocalVariable) {
 		g.Fgenf(w, "%s := %.3v;\n", name, expr)
 
 	}
+}
+
+func (g *generator) genConfigVariable(w io.Writer, v *hcl2.ConfigVariable) {
+	if !g.configCreated {
+		g.Fprintf(w, "%sconfig := config.New(ctx, \"\")\n", g.Indent)
+		g.configCreated = true
+	}
+
+	getType := "Object"
```

Comment:
I don't think this is the right default. I would start with string, and only default to object if the type is `model.Dynamic`. Take a look through the generated aws examples. It looks like every instance of `RequireObject` really just wants strings (or IDs which are a subset of strings):

https://github.com/pulumi-bot/pulumi-aws/compare/ba96ff026..05673f914

Reply:
Yeah I wasn't unsure of this, thanks for your suggestion. I have made the change you suggested and will audit the downstream diffs more closely.

Reply:
> I don't think this is the right default. I would start with string, and only default to object if the type is `model.Dynamic`. Take a look through the generated aws examples. It looks like every instance of `RequireObject` really just wants strings (or IDs which are a subset of strings):
> 
> https://github.com/pulumi-bot/pulumi-aws/compare/ba96ff026..05673f914

Hmm seems like checking for dynamic type is not sufficient - a bunch of the ID types are still being emitted as objects. Wonder if there is a way to do an assignability check. Will update with what I find.

Reply:
> > I don't think this is the right default. I would start with string, and only default to object if the type is `model.Dynamic`. Take a look through the generated aws examples. It looks like every instance of `RequireObject` really just wants strings (or IDs which are a subset of strings):
> > https://github.com/pulumi-bot/pulumi-aws/compare/ba96ff026..05673f914
> 
> Hmm seems like checking for dynamic type is not sufficient - a bunch of the ID types are still being emitted as objects. Wonder if there is a way to do an assignability check. Will update with what I find.

Actually seems like these are showing up as object in the other language generators too. We should probably just cut a separate ticket to track and fix these.